make stringification configureable

Tom-Hirschberger opened the pull request:
Hi,

i do have problems controlling my Shelly plugs with the module cause of the "JSON.stringify" happening before the payload will be send as MQTT message.

With the changes in this pull request the stringification can be controlled with a general configuration option "stringifyPayload" which can be overriden by notifications in their configuration in notiMqttCommands if needed.

The default behavior stays the same.

sergge1 commented:
That is great. Tom, do you want to maintain this repository as an owner? 
I am not longer use the magicmirror, which means that I do not participate in this repository development. Can only accept or reject pull requests. 
That is why I think it is better to transfer it to someone who can keep it alive and up to date.

Tom-Hirschberger commented:
If you like i can take the ownership. I can not guarantee to add a lot of features in the next time but i can take care to keep it compatible with current version of MagicMirror².

How does the transfer process work? Will the URL of the Repository be still available or do the users have to change the origin URL with a git command?

sergge1 commented:
I just tried to move ownership to you. Github returned me with information that you have a repository with the same name, which means that you have to delete your fork, and after that, I will make a move for you. Can you please try this?
The only thing is that address will be changed to Tom-Hirschberger/MMM-MQTTbridge means that if some links 
follow to this repository; they will not work.

Tom-Hirschberger commented:
I removed the fork at this moment. I can give it a second try. Maybe i will post in the forum that the ownership changed and that the users should set the new origin url to keep it up to date.

Tom-Hirschberger commented:
Sorry. Meant you can give it a second try.

sergge1 commented:
Yes, it would be a good idea to inform the users.
I just sent the ownership, GitHub is waiting for your confirmation

sergge1 commented:
seems that old link works as well

Tom-Hirschberger commented:
Thank you very much.
It worked like a charm.
And yes, i got redirected to the new url immediately.

sergge1 commented:
that is. good. Thank you for taking it. 
As I said, I am far away from this project in general, so I think someone involved can take better care of it.
Thank you again! You told me above that you are using Shelly. Do you use them with Home Assistant or as a separate instance?

sergge1 commented:
https://forum.magicmirror.builders/topic/12230/mmm-mqttbridge
i was able to log in to the forum and change the initial topic ;) Now all the references route to your repository :)

Tom-Hirschberger commented:
Perfect.

Basically i use KNX bus in my home. But i added a Node-RED instance connected to a "knxd" installation to control some KNX devices with a Node-RED dashboard.
Since a few months i use a 3D printer i added a camera and led strip to. The led strip is connected to a ESP32 controller and can be switched on and off via mqtt.
I can access my mirror instance via VPN and watch the printers webcam with a MJPEG stream i embedded into the mirror. If the printer finished or if something wents wrong i wanted to abort the printing job by switching off the power of the printer.
So i ended up powering the printer with a Shelly Plug S and add a button to the mirror (with my MMM-TouchButton module) to switch the power off the Shelly Plug S with MQTT.
There is a Node-RED integration of Shelly as well but as i do have a MQTT infrastructure already it was easier this way.

sergge1 commented:
I understood. Very cool setup indeed. :)
I was using MM with Home Assistant, and that is the initial reason for the development of this module. I see you want to implement all MQTT features - like QoS, "last will message" etc - that is an amazing plan!) Having this - the module becomes universal and secure!

sergge1 commented:
Just checked one thing - I changed the LICENCE file, removed myself, and added you to it. GitHub allowed me to do so without a pull request with your approval.
Seems I have some special rights, so just in case - please remove such rights from me, just to be secure that only you can validate pull requests, if any.

Tom-Hirschberger commented:
I finished most of the job today.
Did some refactoring and speed up the notification listening.
Also all cennection (i.e. clean session) , publishing (QoS, retain) and subscription (QoS) are possible now.
Will try the implement a second way of configuration tomorrow and make some last test then the new release is finished.

Tom-Hirschberger commented:
Oh. I will check that. Thanks for the hint

sergge1 commented:
Hoooray!:) Good luck with the tests!
